crates/dekaf/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM lukemathwalker/cargo-chef:latest-rust-1 AS chef
```

Comment:
I understand you're probably using this to more easily develop on Mac, but all other CI pipelines build directly within the Github actions runner environment and then package the binary into a docker container.

The reason we do it this way is for across-the-board consistency of the build environment of our binaries.

Reply:
Alright, I can make that change. It's a deviation from my understanding of the "normal" way to build an image with a `Dockerfile` and instead encoding more of the build logic in github actions's less portable/standard syntax... but consistency is valuable and I'm pretty sure it'll be faster to save/load build caches. Out of curiosity, why do we do it this way in the first place?

Reply:
Quite literally right this moment, @psFried is debugging a subtle weirdness / breakage in our `source-http-ingest` connector which is currently suspected to be an issue in its... wait for it... dockerized build process :laughing: 

Build environments are a *long term* commitment. It's not enough that it builds today. The environment also needs to be continuously updated with version bumps, security fixes, and so on. It's never actually a static environment, as much as the idea of "hermetic builds" is appealing. One day `lukemathwalker` will move on, and the image will not be updated any more, and it will bit rot to an unacceptable degree, and there will be a scramble to get another build set up that usually requires first de-tangling bespoke bits of the legacy one.

In that framing, Github actions environments are **very much** a standardized build environment: an environment used by tons of people, that see continuous delivery of updated compilers, base OS images, and so on.

C.f. Gazette's make-within-a-docker-builder-called-by-make build system which I wrote, and which is "clever", but which is today quite bit-rotted and confounding even to me.

Reply:
I'm happy to make this change and have most of this done and will push it up tmw, but I am curious about understanding your position on this more clearly.

> One day ’lukemathwalker’ will move on, and the image will not be updated any more

This is the point I most clearly understand and agree with. This base image isn't adding all that much ontop of its own base image so I should really just include it here and depend on e.g ’rust:latest’ instead. 

> Build environments are a long term commitment. It's not enough that it builds today. The environment also needs to be continuously updated with version bumps, security fixes, and so on.

💯. If you depend on e.g ’rust:latest’ or even ’rust:1-bullseye’, you should have an always up to date build environment so long as the Rust organization maintains their base images, much in the same way depending on ’ubuntu-latest’ + ’actions-rust-lang/setup-rust-toolchain@v1’ does in GH actions. 
 
> In that framing, Github actions environments are very much a standardized build environment: an environment used by tons of people, that see continuous delivery of updated compilers, base OS images, and so on.

Sure, but I guess I’m not seeing how they’re any better than Dockerfiles? They both let you pull in snippets written and maintained (or not) by other people, they both let you define some series of build steps to run with inputs and outputs, they both let you specify dependencies/environments in such a way that they’re “always up to date” for better (security updates) or worse (subtly breaking changes to things you depend on). 


I would argue that Dockerfiles are better because they’re not nearly as vendor-locked in. I can find all sorts of non-Docker options for building, testing and running Dockerfiles on my machine exactly as they are in CI, whereas with GitHub actions, your only option is to run them inside GH’s infrastructure against whatever set of largely opaque base images you’re offered. I dunno about you, but my build scripts be they Docker or GH actions _never_ work on the first (or second, or third) try and I’d much rather develop them locally than have to awkwardly force-push commits over and over until I get a green checkmark.

Reply:
I'm going to not wade in further to protect my time. Please junk the dockerized build. Make `dekaf` a target of the `linux-binaries` make rule. You can include it into the singular `flow` docker image we push today, or create a new minimal docker image that wraps the pre-built binary.